dev/PowerNotifications/PowerNotifications.h

```diff
@@ -0,0 +1,699 @@
+            {
+                CheckRegistrationAndOrUpdateValue(userPresenceStatusFunc);
+                return static_cast<winrt::Microsoft::ProjectReunion::UserPresenceStatus>(m_cachedUserPresenceStatus);
+            }
+
+            event_token UserPresenceStatusChanged(const PowerEventHandle& handler)
+            {
+                return AddCallback(userPresenceStatusFunc, handler);
+            }
+
+            void UserPresenceStatusChanged(const event_token& token)
+            {
+                RemoveCallback(userPresenceStatusFunc, token);
+            }
+
+            void UserPresenceStatusChanged_Callback(DWORD userPresenceStatus)
+            {
+                m_cachedUserPresenceStatus = userPresenceStatus;
+                RaiseEvent(userPresenceStatusFunc);
+            }
+
+            //SystemSuspend Functions
+            winrt::Microsoft::ProjectReunion::SystemSuspendStatus SystemSuspendStatus()
+            {
+                // Review: Agree with this "only callable after subscription callback" enforcement?
+                if (m_systemSuspendStatus == SystemSuspendStatus::Uninitialized)
+                {
+                    throw winrt::hresult_error(E_FAIL, L"API only callable after a SystemSuspendStatusChanged callback");
+                }
+                return static_cast<winrt::Microsoft::ProjectReunion::SystemSuspendStatus>(m_systemSuspendStatus);
+            }
+
+            event_token SystemSuspendStatusChanged(const PowerEventHandle& handler)
+            {
+                return AddCallback(systemSuspendFunc, handler);
+            }
+
+            void SystemSuspendStatusChanged(const event_token& token)
+            {
+                RemoveCallback(systemSuspendFunc, token);
+            }
+
+            void SystemSuspendStatusChanged_Callback(ULONG PowerEvent)
+            {
+                using namespace winrt::Microsoft::ProjectReunion;
+                if (PowerEvent == PBT_APMSUSPEND)
+                {
+                    m_systemSuspendStatus = SystemSuspendStatus::Entering;
+                    RaiseEvent(systemSuspendFunc);
+                }
+                else if (PowerEvent == PBT_APMRESUMEAUTOMATIC)
+                {
+                    m_systemSuspendStatus = SystemSuspendStatus::AutoResume;
+                    RaiseEvent(systemSuspendFunc);
+                }
+                else if (PowerEvent == PBT_APMRESUMESUSPEND)
+                {
+                    m_systemSuspendStatus = SystemSuspendStatus::ManualResume;
+                    RaiseEvent(systemSuspendFunc);
+                }
+                // Resetting the value after the callback
+                m_systemSuspendStatus = SystemSuspendStatus::Uninitialized;
+            }
+
+        };
+    }
+
+    namespace implementation
+    {
+        // EffectivePowerMode variables
+        static std::atomic<ULONG> g_powerModeVersion = NULL;
```

Comment:
Is `= NULL` necessary? Can this just be `static std::atomic<ULONG> g_powerModeVersion;` ?

Reply:
Changed.